### ISSUE TITLE
invalid JSON dump

### ISSUE THREAD

**Issue by dAnjou** [noirotm/flvmeta#10]
Hi,

this is probably very easy to fix for someone who knows C and flvmeta's code a little bit.

We're using Red5 to record audio/video streams and the problem described here is probably also a problem there. Anyway, Red5 writes the following file which is basically only the header because something crashed along the way. And I'm trying to use flvmeta to check in such cases:
```
$ flvmeta --dump --json foo-v0.flv  # JSON formatted for your convenience
{
  "duration": 0,
  "server": "Red5",
  "creationdate": "Sun Apr 19 16:39:36 UTC 2015",
  "videocodecid": "avc1",
  "canSeekToEnd": true,
  "videodatarate": -nan,
  "noaudiocodec": 0
}
```
As you can see, there's invalid JSON: `"videodatarate": -nan`.

Here's the file so that you can check yourselves (base64-encoded so that I can paste it here):
```
$ base64 foo-v0.flv
RkxWAQEAAAAJAAAAABIAALgAAAAAAAAAAgAKb25NZXRhRGF0YQgAAAAAAAhkdXJhdGlvbgAAAAAA
AAAAAAAGc2VydmVyAgAEUmVkNQAMY3JlYXRpb25kYXRlAgAcU3VuIEFwciAxOSAxNjozOTozNiBV
VEMgMjAxNQAMdmlkZW9jb2RlY2lkAgAEYXZjMQAMY2FuU2Vla1RvRW5kAQEADXZpZGVvZGF0YXJh
dGUA//gAAAAAAAAADG5vYXVkaW9jb2RlYwAAAAAAAAAAAAAACQAAAMMJAAAuAAAAAAAAABcAAAAA
AU0AH//hABpnTUAfllQFAX/LgKhAAAADAEAAAAynaEiagAEABGjONSAAAAA5
```

Thanks.

**Comment by noirotm** [comment]
Thanks for the bug report.
This must mean that the metadata is malformed somehow, and sends an invalid floating point value.
I will analyze this and keep you informed.

**Comment by noirotm** [comment]
Fixed in commit c3d168b.

The new behaviour is now to print a `null` value instead of an invalid string representation of the floating point value. This is based on recommendations found in the ECMA standard, and the javascript behaviour in this case.
I was not really aware of this problem before, but thanks to this bug report, I have now read a few articles that emphasize this apparent oversight of the JSON standard.
JSON uses decimal numbers, and while not enforcing an IEEE 754 representation, is strongly tied with it. At the same time, it does not allow the representation of `infinity` and `NaN` values, which are standard values in IEEE 754.

This bug fix introduces an optional dependence to a C99 function (isfinite). CMake will detect its presence and eventually use an alternative mechanism if not found.

**Comment by dAnjou** [comment]
Great, thanks.
